src/lorarmesh.cpp

```diff
@@ -14,38 +29,71 @@ LoraMesh::LoraMesh(DataQueue<message_t>* txQueue,
   this->txQueue = txQueue;
   this->rxQueue = rxQueue;
   this->router = router;
+  
+  Layer1 = new Layer1Class();
+  Layer1->setPins(LORA_CS, LORA_RST, LORA_IRQ);
+  Layer1->setTxPower(TX_POWER);
+  Layer1->setLoRaFrequency(LORA_FREQ);
+  int newTry = 0;
+  while (!Layer1->init() or newTry < 30) 
+  {
+    newTry++;
+  }
+  if (newTry < 30){
+    LL2 = new LL2Class(Layer1);  
+    LL2->setInterval(10000); // set to zero to disable routing packets
+    LL2->init();
+  }else{
+    delay(500);
+    ESP.restart();
+  }
+  
   xTaskCreatePinnedToCore(task, "LoraMesh", 10000, this, 0, &taskHandle,
                           LORA_TASKS_CORE);
+  
 }
 
 void LoraMesh::transmit() {
   auto message = txQueue->poll();
+
   if (message == nullptr)
     return;
-
+  
+  LL2->daemon(); 
   if (message->type == CONTROL_MESSAGE) {
-    // TODO broadcast message over LoRaLayer2
-  } else if (message->type == DATA_MESSAGE) {
+    int datagramsize = 0;
+    struct Datagram datagram;
+    memcpy(datagram.destination, BROADCAST_NODES, ADDR_LENGTH); 
+    datagram.type = message->type;
+    memcpy(datagram.message, &message->data.control, sizeof(control_data_t));
+    datagramsize = DATAGRAM_HEADER;  
+    datagramsize += sizeof(control_data_t); 
+    
+    LL2->writeData(datagram, datagramsize);
+    }else if (message->type == DATA_MESSAGE) {
     auto destinationAddr = router->getGatewayAddress();
-    if (destinationAddr == 0)
-      goto freeData;
-
-    // TODO send to destinationAddr through LoRaLayer2
+    if (destinationAddr > 0){
+      int datagramsize = 0;
+      struct Datagram datagram;
+      datagram.type = message->type;
+      memcpy(datagram.destination, &destinationAddr, ADDR_LENGTH); 
+      memcpy(datagram.message, message->data.layer2.payload, message->data.layer2.length);
+      datagramsize = DATAGRAM_HEADER;  
+      datagramsize += message->data.layer2.length;
+      
+      LL2->writeData(datagram, datagramsize);
+    }
+    
   }
-
-freeData:
+  
   if (message->type == DATA_MESSAGE) {
     free(message->data.layer2.payload);
   }
   free(message);
 }
 
 void LoraMesh::receive() {
-  message_t* message = nullptr;  // TODO read actual data from LoRaLayer2
```

Comment:
Do not change the receive method.

Reply:
`void LoraMesh::receive() {
  message_t* message = nullptr;  // TODO read actual data from LoRaLayer2
  if (message == nullptr)
    return;

  if (message->type == CONTROL_MESSAGE) {
    router->handleControlMessage(message);
  }
}`